Gemfile

```diff
@@ -3,19 +3,19 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.2.2'
+ruby '3.3.0'
 
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
-gem 'rails', '~> 7.0.8'
+gem 'rails', '~> 7.1.2'
```

Comment:
At first, the change from Rails from 7.0 to 7.1 was due to the ability to create a Dockerfile easily, but that did not work. I did that regardless.

Notable changes I might go with in this project are [Active Record Normalisers](https://guides.rubyonrails.org/7_1_release_notes.html#add-activerecord-base-normalizes) (a syntax sugar for before_validation callback to assign or change an attribute), [Async Queries that return something other than a Relation](https://guides.rubyonrails.org/7_1_release_notes.html#active-record-api-for-general-async-queries), and possibly the [perform_all_later](https://guides.rubyonrails.org/7_1_release_notes.html#add-perform-all-later-to-enqueue-multiple-jobs-at-once) addition.